### DISCUSSION TITLE
Is there any chance that flv videos embeded in swf files be able to play ?

### DISCUSSION THREAD

**Discussion by BracerJack** [ruffle-rs/ruffle#683] [General]
I am totally not expecting this to be implemented but thought I might ask to see if this is part of the target that is something to look forward to or something that is to be surrender to history forever.

**Comment by Herschel** [comment]
(Our repo was added to the GitHub Discussions beta, so I've moved this from Issues to Discussions as it is not a specific bug report, and to test out Discussions, thanks!)

Supporting video is definitely within the scope of the project. The main issues are a) deciding how to decode certain formats on various platforms (possibly leveraging FFmpeg and/or the browser's built-in decoder on web, when available), and b) dealing with potentially patent-encumbered codecs such as H.264 (You can read #590 for more info).

As for a timeframe, video support is not very high-priority at the moment, so it will probably be months before any video support is added; certainly the simpler codecs such as Screen Video will be supported before the more complex ones.

**Reply by 3TOON** [reply]
First, thanks a lot for all your efforts for keeping Flash content alive!

Regarding the video, any kind of support would be a game changer!
I have tried to import sequences of images to simulate video, but it is not usable... Much too heavy, and when the file is played online, after a few seconds, it just breaks...
I have used videos in Flash to make personnalized content, and it was just fantastic!
Doing the same thing with HTML5 implies creating a new video file each time it is personnalized... For some, it is a great and very lucrative business, unfortunately for me, it's a "no go". For one of my clients, my personnalized Flash cartoon was sent to 300.000 people. It was relatively easy and cheap. Imagine creating 300.000 video files to obtain the same result in HTML5... Just insane!
Please consider supporting video in Ruffle, it would change my life, my work and what the Web could be, again!

**Comment by vladislavkoz** [comment]
Hello guys. Any updates on this? I want to recover a video game with a lot of FLV files and I'm getting the following error. As I understand I can't use FLV files. As I understand support for FLV files is not added yet? Could you please clarify which video formats could be used?

![image](https://user-images.githubusercontent.com/40685761/160371650-fb12f8d6-0e26-45ee-b71d-e7cb38ecb459.png)

FLV file meta data:
![image](https://user-images.githubusercontent.com/40685761/160389530-5eb6b885-4c81-433a-9b6a-b6de9f2abb1c.png)

**Reply by n0samu** [reply]
Loading external FLV files is still not supported. Ruffle does support the VP6 codec for videos embedded inside SWF files, though. The other codecs supported by Ruffle are H.263 and Screen Video.

**Comment by Angelowilliam1** [comment]
Is there a possibility that I can see flv files embedded in swf files? How is the quest to fix it going?

**Reply by n0samu** [reply]
If you just want to watch a video embedded in an SWF file, there's a good chance that VLC Media Player can play it, simply by opening the SWF file. You can also extract the video data to an FLV file using [JPEXS Flash Decompiler](https://github.com/jindrapetrik/jpexs-decompiler/).

**Reply by BracerJack** [reply]
That is wonderful, thank you n0samu.

**Reply by Angelowilliam1** [reply]
I have a site that works in localhost, within the site pages there are video files with flv extension incorporated by swf. I would like to see the site as it did when Adobe Flash Player was around. Thanks for your interest. I write from Italy and use google translate.

**Reply by n0samu** [reply]
If you have a Windows PC, the easiest option to view the website is to use this browser with Flash Player included: https://archive.org/details/basilisk-portable-with-flash

**Reply by Angelowilliam1** [reply]
Thanks for helping me, my Flash site works (embed flv videos with swf). I hope as soon as possible also Ruffle can make it with Google Chrone. Thanks for everything. Hello

**Comment by Lord-McSweeney** [comment]
Loading external FLV files is now supported, along with audio and seeking for those FLV files- but streaming playback of FLVs and MP4s are still not yet supported.
